pshtt.py

```diff
@@ -520,7 +548,11 @@ def is_downgrades_https(domain):
     else:
         canonical_https = https
 
-    return (supports_https and canonical_https.redirect_immediately_to_http)
+    return (
+        supports_https and
+        canonical_https.redirect_immediately_to_http and
+        (not canonical_https.redirect_immediately_to_external)
+    )
```

Comment:
Our logic was differing from `site-inspector` here in a way I failed to notice at first -- we only want to ding a domain for downgrading HTTPS if it downgrades *internally*. If some redirect domain (e.g. `https://redirect.gov`) is doing all it can to enforce HTTPS for itself, but has to redirect to some other domain that doesn't have HTTPS set up yet, that's the fault of the domain it's redirecting to, and will be surfaced in that domain or subdomain's report instead of the redirector's.

So this fix above modifies the downgrade detection logic to only care if the canonical endpoint downgrades to HTTP within the domain.